extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/pages/databaseSettingsPage.ts

```diff
@@ -128,9 +149,33 @@ export class DatabaseSettingsPage extends BasePage {
 		this._endIpAddressTextRow = this.wizard.createFormRowComponent(view, constants.EndIpAddressLabel, '', this._endIpAddressTextbox, true);
 	}
 
-	private createFirewallNameText(view: azdata.ModelView) {
+	private validateFirewallNameText(firewallname: string | undefined): boolean {
+		if (firewallname) {
+			if (/^\d+$/.test(firewallname)) {
+				return false;
+			}
+			else if (firewallname.length < 1 || firewallname.length > 100) {
+				return false;
+			}
+			else if (/[\\\/"\'\[\]:\|<>\+=;\?\*@\&,\{\} ]/g.test(firewallname)) {
+				return false;
+			}
+			else if (/[A-Z]/g.test(firewallname)) {
+				return false;
+			}
+			else {
+				return true;
+			}
+		} else {
+			return false;
+		}
+	}
 
-		this._firewallRuleNameTextbox = view.modelBuilder.inputBox().component();
+	private createFirewallNameText(view: azdata.ModelView) {
+		this._firewallRuleNameTextbox = view.modelBuilder.inputBox().withProperties(<azdata.InputBoxProperties>{
```

Comment:
I'm not really sure what this sentence is saying the valid values are supposed to be. It's usually a lot clearer if you say what IS allowed - is this text from documentation somewhere or something?

It also seems like the test is incorrect based on this - the message here claims that the firewall rule can't contain upper case letters but you only check that the first letter isn't upper case